### PR TITLE
Code coverage in Maven using JaCoCo

### PR DESCRIPTION
**Pull request by cobratbq**
Adds maven plugin JaCoCo for generating a code coverage report based on the run unit tests. The report is also included into the `mvn site` report if it is generated.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,6 +177,16 @@
                     </rulesets>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>0.7.2.201409121644</version>
+                <configuration>
+                    <excludes>
```
Comment (languitar) [inline pom.xml]: I would leave this useless section out as long as it is not required.

**Comment by cobratbq** [comment]
Removed empty configuration section.

**Comment by languitar** [comment]
Looks good to me but I would squash everything down to a single commit before putting it onto master.

**Comment by cobratbq** [comment]
Squashed down and ready to push :-)

**Comment by languitar** [comment]
I would push it if I had commit access ;)

**Comment by cobratbq** [comment]
In that case, I appreciate the vote of confidence :+1:

**Pull request closed without merging**